src/dmqproto/node/neo/request/core/IRequestHandlerRequest.d

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+
+    Base class for requests. Implements IRequestHandler and leaves one method,
+    `handle`, to be implemented by the subclass.
+
+    Copyright:
+        Copyright (c) 2016-2018 sociomantic labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module dmqproto.node.neo.request.core.IRequestHandlerRequest;
+
+import swarm.neo.node.IRequestHandler;
+
+/// ditto
+abstract class IRequestHandlerRequest: IRequestHandler
+{
+    import swarm.neo.node.RequestOnConn;
+    import dmqproto.node.neo.request.core.IRequestResources;
+    import ocean.transition;
+
+    /// Request-on-conn of this request handler.
+    private RequestOnConn connection;
+
+    /// Acquired resources of this request.
+    private IRequestResources resources;
+
+    /***************************************************************************
+
+        The payload of the initial message.
+        This array references an acquired buffer so do not change the length.
+
+    ***************************************************************************/
+
+    private Const!(void)[] msg_payload;
+
+    /***************************************************************************
+
+        Stores the request-on-conn and request resource acquirer, to be passed
+        to `handle`.
+
+        Params:
+            connection = request-on-conn in which the request handler is called
+            resources = request resources acquirer
+
+    ***************************************************************************/
+
+    public void initialise ( RequestOnConn connection, Object resources_object )
+    {
+        this.connection = connection;
+        this.resources = cast(IRequestResources)resources_object;
+        assert(this.resources !is null);
+    }
+
+    /***************************************************************************
+
+        Copies `init_payload` into an acquired buffer, to be passed to `handle`.
+
+        Params:
+            init_payload = initial message payload read from the connection
+
+    ***************************************************************************/
+
+    public void preSupportedCodeSent ( Const!(void)[] init_payload )
+    {
+        void[]* buf = this.resources.getVoidBuffer();
```

Comment:
We discussed this and concluded it is the best approach:
- In practice most of the request handlers (all in the DMQ) need to pass a copy of some string from `preSupportedCodeSent` to `postSupportedCodeSent` so the whole `init_payload` could be passed as well.
- The `IRequestHandler` interface would be simplified to one method, `handle`. (It is still good to use an interface because it can easily be extended.)
- Passing the results from `preSupportedCodeSent` to `postSupportedCodeSent` is ugly, it introduces context-specific class variables, which should better be local variables in a `handle` method.